crates/bevy_ecs/src/storage/blob_vec.rs

```diff
@@ -375,8 +385,8 @@ mod tests {
     #[test]
     fn resize_test() {
         let item_layout = Layout::new::<usize>();
-        let drop = drop_ptr::<usize>;
-        let mut blob_vec = unsafe { BlobVec::new(item_layout, drop, 64) };
+        // usize doesn't need dropping
```

Comment:
This changes the behavior though. Maybe keep the drop function here and add a new test with `None` as drop?

Reply:
It's not clear to me how this has different behaviour. Could you clarify this please?

Either we 'drop' the usize values by calling a function pointer, which does nothing, or we do nothing without calling a function pointer.

Reply:
This change means you are not testing the same as before. The drop path is no longer tested. There should be no observable difference if there are no bugs, but a test is meant to check that there are indeed no bugs.

Reply:
This test isn't trying to test dropping though, it's testing that resizing upon `push` works. If you like, this is already testing a no-op drop, and now continues to test a no-op drop.
The case where `Drop` is `Some` is tested thoroughly below; this strictly tests more paths.